kube/Cargo.toml

```diff
@@ -50,8 +50,8 @@ default-features = false
 [dev-dependencies]
 tokio = { version = "1.14.0", features = ["full"] }
 futures = "0.3.17"
+garde = { version = "0.11.2", default-features = false, features = ["derive"] }
```

Comment:
if this only does client-side validation now, we should probably just take it out and leave `garde` as an example nicety as it currently is in `crd_api.rs`.

the elaborating doc comments in kube-derive are good, but we can probably remove references to `garde` in the actual doc examples for one less crate to build.